lib/rspec/matchers/built_in/be_instance_of.rb

```diff
@@ -14,7 +14,21 @@ def description
       private
 
         def match(expected, actual)
-          actual.instance_of? expected
+          if actual_object_respond_to?(actual, :instance_of?)
+            actual.instance_of?(expected)
+          else
+            raise ::ArgumentError, error_message
+          end
+        end
+
+        def actual_object_respond_to?(actual, method)
+          ::Kernel.instance_method(:respond_to?).bind(actual).call(method)
+        end
+
+        def error_message
```

Comment:
Given this is now a raise and reused, can we inline it?

Reply:
Yes, I will update.

Reply:
@JonRowe 
You mean that `#actual_object_respond_to?` should be inline?
If not, can you provide me more details?

Reply:
```
  raise ArgumentError, "The #{matcher_name} matcher requires that " \
                                      "the actual object responds to #instance_of? method " \
                                      "but it does not respond to the method."
```

Reply:
Thank you!

Reply:
Done.